### PR TITLE
docs: add gerrit integration

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: "Gerrit Hook"
+weight: 7
+---
+
+Do you want to trigger a workflow from a gerrit event? This kind of hook is for you.
+
+You have to:
+
+* link your project to a Gerit Server, on Advanced Section
```
Comment (fsamin) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* link your project to a Gerrit Server, on Advanced Section
```

File: docs/content/docs/integrations/gerrit.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+---
+title: Gerrit Repository Manager
+main_menu: true
+card: 
+  name: repository-manager
+---
+
+The Gerrit Repository Manager Integration have to be configured on your CDS by a CDS Administrator.
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
The Gerrit Repository Manager integration have to be configured on your CDS by a CDS Administrator.
```

File: docs/content/docs/integrations/gerrit.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+This integration allows you to link a Git Repository hosted by Gerrit
+to a CDS Application.
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
to a CDS application.
```

File: docs/content/docs/integrations/gerrit.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+ - An Administrator User ( with SSH KEY ), to get event from Gerrit Server
+ - An User on gerrit ( with httpPassword ), to comment changes with workflow result
+ 
+
+### Complete CDS Configuration File
+
+```yaml
+ [vcs.servers]
+
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
```

File: docs/content/docs/integrations/gerrit.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ - An User on gerrit ( with httpPassword ), to comment changes with workflow result
+ 
+
+### Complete CDS Configuration File
+
+```yaml
+ [vcs.servers]
+ 
+     [vcs.servers.gerrit]
+
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
```

File: docs/content/docs/integrations/gerrit.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+```yaml
+ [vcs.servers]
+ 
+     [vcs.servers.gerrit]
+ 
+       # URL of Gerrit
+       url = "http://localhost:9080"
+ 
+       [vcs.servers.gerrit.gerrit]
+
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md (hunk 1 above)
Comment (bnjjj) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* link your project to a Gerrit Server, on Advanced Section
```

File: docs/content/docs/integrations/gerrit.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+       [vcs.servers.gerrit.gerrit]
+ 
+         # Disable event listener
+         disableGerritEvent = false
+ 
+         # SSH port of gerrit
+         sshport = 29418
+ 
+         [vcs.servers.gerrit.gerrit.EventStream]
+
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+* link an application to a gerrit repository
```
Comment (bnjjj) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* link an application to a Gerrit repository
```

File: docs/content/docs/integrations/gerrit.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+         [vcs.servers.gerrit.gerrit.EventStream]
+ 
+           # Private key of the user who access to gerrit event stream
+           privateKey = "XXXXXXX"
+ 
+           # User to access to gerrit event stream
+           user = "admin"
+ 
+         [vcs.servers.gerrit.gerrit.Reviewer]
+
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+* add a Gerrit Hook on the root pipeline, this pipeline have the application linked in the [context]({{< relref "/docs/concepts/workflow/pipeline-context.md" >}})
+
+With this hooks, you will have access to specific variables:
```
Comment (bnjjj) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
With this hook, you will have access to specific variables:
```

File: docs/content/docs/integrations/gerrit.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+           user = "admin"
+ 
+         [vcs.servers.gerrit.gerrit.Reviewer]
+ 
+           # Http Password of the user that comment changes
+           token = "XXXXXXXXXX"
+ 
+           # User that review changes
+           user = "Verifier"
+
```
Comment (fsamin) [inline docs/content/docs/integrations/gerrit.md]:
```suggestion
```

File: docs/content/docs/integrations/gerrit.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+The Gerrit Repository Manager Integration have to be configured on your CDS by a CDS Administrator.
+
+This integration allows you to link a Git Repository hosted by Gerrit
+to a CDS Application.
+
+This integration enables some features:
+
+ - [Gerrit Hooks]({{<relref "/docs/concepts/workflow/hooks/gerrit.md" >}})
+ - Easy to use action [CheckoutApplication]({{<relref "/docs/actions/builtin-checkoutapplication.md" >}}) and [GitClone]({{<relref "/docs/actions/builtin-gitclone.md">}}) for advanced usage
+ - Send comments on your Pull-Requests when a workflow is failed
```
Comment (yesnault) [inline docs/content/docs/integrations/gerrit.md]: and vote, no?

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+* gerrit.change.id: ID of the change
```
Comment (yesnault) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* `gerrit.change.id`: ID of the change
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+* gerrit.change.url: URL of the change
```
Comment (yesnault) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* `gerrit.change.url`: URL of the change
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+* gerrit.change.status: Status of the change
```
Comment (yesnault) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* `gerrit.change.status`: Status of the change
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+* gerrit.change.branch: Destination branch of the change
```
Comment (yesnault) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* `gerrit.change.branch`: Destination branch of the change
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+* gerrit.ref.name: Full reference name within project
```
Comment (yesnault) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* `gerrit.ref.name`: Full reference name within project
```

File: docs/content/docs/concepts/workflow/hooks/gerrit.md [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+* gerrit.change.ref: Git reference of the change
```
Comment (yesnault) [inline docs/content/docs/concepts/workflow/hooks/gerrit.md]:
```suggestion
* `gerrit.change.ref`: Git reference of the change
```

**Pull request merged**